pandas/core/computation/pytables.py

```diff
@@ -239,7 +239,8 @@ def stringify(value):
             if conv_val not in metadata:
                 result = -1
             else:
-                result = metadata.searchsorted(conv_val, side="left")
+                # Find the index of the first match of conv_val in metadata
+                result = np.where(metadata == conv_val)[0][0]
```

Comment:
```suggestion
                result = np.flatnonzero(metadata == conv_val)[0]
```
Also is it possible to know if metadata is sorted ahead of time so we can use `searchsorted`? it will be much faster in that case